core/src/main/java/org/elasticsearch/action/support/TransportAction.java

```diff
@@ -89,6 +99,28 @@ public void onFailure(Throwable e) {
         return task;
     }
 
+    public final Task execute(Request request, final TaskListener<Response> listener) {
```

Comment:
Reindex needs this so it can log something when a `wait_for_completion=false` reindex request comes in. This is exactly how it works in master.